packages/agent/src/certificate.ts

```diff
@@ -216,44 +215,6 @@ export class Certificate {
     return this.lookup([label]);
   }
 
-  public cache_node_keys(root_key?: Uint8Array): SubnetStatus {
-    const tree = this.cert.tree;
-    let delegation = this.cert.delegation;
-    // On local replica, with System type subnet, there is no delegation
-    if (!delegation && typeof root_key !== 'undefined') {
-      delegation = {
-        subnet_id: Principal.selfAuthenticating(root_key).toUint8Array(),
-        certificate: new ArrayBuffer(0),
-      };
-    }
-    // otherwise use default NNS subnet id
-    else if (!delegation) {
-      delegation = {
-        subnet_id: Principal.fromText(
-          'tdb26-jop6k-aogll-7ltgs-eruif-6kk7m-qpktf-gdiqx-mxtrf-vb5e6-eqe',
-        ).toUint8Array(),
-        certificate: new ArrayBuffer(0),
-      };
-    }
-    const nodeTree = lookup_path(['subnet', delegation?.subnet_id as ArrayBuffer, 'node'], tree);
-    const nodeForks = flatten_forks(nodeTree as HashTree) as HashTree[];
-    nodeForks.length;
-
-    this.#nodeKeys = nodeForks.map(fork => {
-      const derEncodedPublicKey = lookup_path(['public_key'], fork[2] as HashTree) as ArrayBuffer;
-      if (derEncodedPublicKey.byteLength !== 44) {
-        throw new Error('Invalid public key length');
-      } else {
-        return toHex(derEncodedPublicKey);
-      }
-    });
-
-    return {
-      subnetId: Principal.fromUint8Array(new Uint8Array(delegation.subnet_id)).toText(),
-      nodeKeys: this.#nodeKeys,
-    };
-  }
-
   private async verify(): Promise<void> {
```

Comment:
Are we going to disable the freshness check for delegation in this PR? If so, I guess it should be some changes in this function, right?

Reply:
verify is called as part of Certificate.create which is called on the certificate in the delegation.
verify checks that /time is past now - 5 min and before now + 5 min.

The IC returns root certificates which are several days or weeks old. This validation would fail. See [FOLLOW-1294 ](https://dfinity.atlassian.net/browse/FOLLOW-1294 )

Reply:
The code is a little confusing, but inside of a standard `certificate`, there are two `certificates`. The main one, and the one contained inside the `delegation`. on line 263, we set a special maxAgeInMinutes of 30 days, which hasn't seemed to fail against the mainnet in practice. We could extend this as far as we want though

https://github.com/dfinity/agent-js/blob/a899a3cadc92430071dce6dacfac1333be255e38/packages/agent/src/certificate.ts#L263

Reply:
@ShuoWangNSL could you check whether the behavior is/should be aligned between js and rs.
Atm, I think js checks the cert against 5min **window** unless it's a cert inside a delegation where it's 30 days.
rs only checks the cert is not older than 5 min and does not check the time on certs found in delegations.

Eventually, all certs should use the same 5 min windows (see [FOLLOW-692](https://dfinity.atlassian.net/browse/FOLLOW-692))

Reply:
As we discussed this morning, I agreed to remove the check. In code, this is done by setting the maxAge to `Infinity`